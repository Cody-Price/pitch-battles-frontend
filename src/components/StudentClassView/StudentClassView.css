.student-class-view-wrapper {
  font-family: "chronotype";
  min-height: 100vh;
  display: flex;
  align-items: center;
  position: relative;
}
.student-class-view {
  animation: slide-class-view 0.3s linear;
  -webkit-animation: slide-class-view 0.3s linear;
  -moz-animation: slide-class-view 0.3s linear;
  -o-animation: slide-class-view 0.3s linear;
  background: url("../Game/images/char-dash-backer.png");
  width: 800px;
  height: 600px;
  background-size: cover;
  image-rendering: pixelated;
  margin: auto;
  position: relative;
  z-index: 1000;
  box-shadow: 2px 2px 2px rgba(0, 0, 0.3);
  text-align: start;
}
.student-class-view-header {
  margin-top: 3rem;
  margin-left: 1rem;
}

.class-view-back-btn {
  position: absolute;
  right: 3rem;
  top: 3rem;
  background: #4776bb;
  box-shadow: 2px 2px 2px#000;
  border: none;
}

.class-view-back-btn:hover {
  cursor: pointer;
}

.class-view-back-btn:active {
  box-shadow: none;
}

.class-view-back-btn:focus {
  outline: none;
}

.student-class-view-class-name,
.student-class-view-teacher-name {
  margin-left: 2rem;
}

.student-class-view-class-name {
  margin-bottom: 0.5rem;
}

.student-class-view-teacher-name {
  margin-top: 0;
  margin-bottom: 5rem;
  font-size: 1.4rem;
}

.class-view-fastest-times-row {
  display: flex;
  justify-content: space-around;
  margin: 0 2rem;
}

.class-view-fastest-times-label {
  font-size: 1.8rem;
  text-align: center;
  margin-bottom: 0;
}

.class-view-fastest-time {
  margin: 0 1rem;
  text-align: center;
  font-size: 1.2rem;
  position: relative;
}

.most-games-badges-row {
  display: flex;
  justify-content: space-around;
  font-size: 1.4rem;
  text-align: center;
  margin-top: 4rem;
}

.most-games-badges-label {
  margin-bottom: 0.5rem;
}

.most-games.game-badge {
  position: relative;
}

.most-badges {
  position: relative;
}

@keyframes slide-class-onboarding {
  0% {
    right: 4000px;
  }
  100% {
    right: 0;
  }
}

@-webkit-keyframes slide-class-onboarding {
  0% {
    right: 4000px;
  }
  100% {
    right: 0;
  }
}

@-moz-keyframes slide-class-onboarding {
  0% {
    right: 4000px;
  }
  100% {
    right: 0;
  }
}

@-o-keyframes slide-class-onboarding {
  0% {
    right: 4000px;
  }
  100% {
    right: 0;
  }
}

.avatar-class-view {
  margin: auto;
  height: 60px;
  width: 33px;
  position: absolute;
  left: 0;
  right: 0;
  transition: 0.2s;
  bottom: -3rem;
}

.avatar-class-view.one {
  background: url(../Game/pcs/char1/stand.png);
  image-rendering: pixelated;
  background-size: cover;
  background-repeat: no-repeat;
}

.avatar-class-view.two {
  background: url(../Game/pcs/char2/stand.png);
  image-rendering: pixelated;
  background-size: cover;
  background-repeat: no-repeat;
}

.avatar-class-view.three {
  background: url(../Game/pcs/char3/stand.png);
  image-rendering: pixelated;
  background-size: cover;
  background-repeat: no-repeat;
  width: 35px;
}

.avatar-class-view.four {
  background: url(../Game/pcs/char4/stand.png);
  image-rendering: pixelated;
  background-size: cover;
  background-repeat: no-repeat;
}

.avatar-class-view.five {
  background: url(../Game/pcs/char5/stand.png);
  image-rendering: pixelated;
  background-size: cover;
  background-repeat: no-repeat;
  width: 37px;
}

.avatar-class-view.six {
  background: url(../Game/pcs/char6/stand.png);
  image-rendering: pixelated;
  background-size: cover;
  background-repeat: no-repeat;
  width: 37px;
}

.avatar-class-view.nine {
  background: url(../Game/pcs/char9/stand.png);
  image-rendering: pixelated;
  background-size: cover;
  background-repeat: no-repeat;
  width: 36px;
  height: 60px;
}

.avatar-class-view.ten {
  background: url(../Game/pcs/char10/stand.png);
  image-rendering: pixelated;
  background-size: cover;
  background-repeat: no-repeat;
  width: 42px;
  height: 60px;
}

.avatar-class-view.eleven {
  background: url(../Game/pcs/char11/stand.png);
  image-rendering: pixelated;
  background-size: cover;
  background-repeat: no-repeat;
}

.avatar-class-view.twelve {
  background: url(../Game/pcs/char12/stand.png);
  image-rendering: pixelated;
  background-size: cover;
  background-repeat: no-repeat;
}

.avatar-class-view.seven {
  background: url(../Game/pcs/char7/stand.png);
  image-rendering: pixelated;
  background-size: cover;
  background-repeat: no-repeat;
  width: 42px;
  height: 62px;
}

.avatar-class-view.eight {
  background: url(../Game/pcs/char8/stand.png);
  image-rendering: pixelated;
  background-size: cover;
  background-repeat: no-repeat;
  width: 43px;
  height: 62px;
}

.avatar-class-view.thirteen {
  background: url(../Game/pcs/char13/stand.png);
  image-rendering: pixelated;
  background-size: cover;
  background-repeat: no-repeat;
  width: 35px;
}

.avatar-class-view.fourteen {
  background: url(../Game/pcs/char14/stand.png);
  image-rendering: pixelated;
  background-size: cover;
  background-repeat: no-repeat;
  width: 33px;
}
