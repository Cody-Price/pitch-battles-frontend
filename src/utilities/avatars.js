const avatars = [
  "one",
  "one",
  "two",
  "three",
  "four",
  "five",
  "six",
  "seven",
  "eight",
  "nine",
  "ten",
  "eleven",
  "twelve",
  "thirteen",
  "fourteen"
];

export default avatars;
