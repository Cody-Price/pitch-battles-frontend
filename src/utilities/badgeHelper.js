export const badgeHelper = [
  { name: "level_one", badge: "level one: completed", rank: 2 },
  { name: "level_one_perfect", badge: "level one: perfect", rank: 3 },
  { name: "level_two", badge: "level two: completed", rank: 4 },
  { name: "level_two_perfect", badge: "level two: perfect", rank: 6 },
  { name: "level_three", badge: "level three: completed", rank: 7 },
  { name: "level_three_perfect", badge: "level three: perfect", rank: 9 },
  { name: "level_four", badge: "level four: completed", rank: 11 },
  { name: "level_four_perfect", badge: "level four: perfect", rank: 12 },
  { name: "all_perfect", badge: "all levels: perfect", rank: 14 },
  { name: "five_games", badge: "play 5 games", rank: 1 },
  { name: "ten_games", badge: "play 10 games", rank: 5 },
  { name: "twenty_games", badge: "play 20 games", rank: 8 },
  { name: "fifty_games", badge: "play 50 games", rank: 10 },
  { name: "one_hundred_games", badge: "play 100 games", rank: 13 },
  { name: "two_hundred_games", badge: "play 200 games", rank: 15 },
  { name: "five_hundred_games", badge: "play 500 games", rank: 16 },
  { name: "one_thousand_games", badge: "play 1000 games", rank: 17 }
];
