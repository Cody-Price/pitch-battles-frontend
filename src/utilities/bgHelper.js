export const bgHelper = [
  "one",
  "two",
  "three",
  "four",
  "five",
  "six",
  "seven",
  "eight"
];
