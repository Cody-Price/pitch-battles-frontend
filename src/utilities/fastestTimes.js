export const fastestTimes = user => {
  console.log(user);
};
