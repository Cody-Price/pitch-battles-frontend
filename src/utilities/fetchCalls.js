export const login = async body => {
  const url = "https://pitch-battles-api.herokuapp.com/login";

  const options = {
    body: JSON.stringify(body),
    method: "POST",
    mode: "cors",
    headers: {
      "Content-Type": "application/json"
    }
  };

  const response = await fetch(url, options);
  return await response.json();
};

export const signUp = async body => {
  const url = "https://pitch-battles-api.herokuapp.com/api/v1/users";

  const options = {
    body: JSON.stringify(body),
    method: "POST",
    mode: "cors",
    headers: {
      "Content-Type": "application/json"
    }
  };

  const response = await fetch(url, options);
  return await response.json();
};

export const userFetch = async webToken => {
  const url = "https://pitch-battles-api.herokuapp.com/api/v1/dashboard/";

  const options = {
    method: "GET",
    mode: "cors",
    headers: {
      Authorization: `bearer ${webToken}`,
      "Content-Type": "application/json"
    }
  };

  console.log(options);

  const response = await fetch(url, options);
  return await response.json();
};

export const postGameUserUpdate = async (update, webToken) => {
  const url = "https://pitch-battles-api.herokuapp.com/api/v1/users";

  const body = update;

  const options = {
    body: JSON.stringify(body),
    method: "POST",
    mode: "cors",
    headers: {
      "Content-Type": "application/json"
    }
  };
  return;
};

export const forgotMyPasswordCall = async email => {
  // const url
  // return response
  return Promise.resolve(true);
};
