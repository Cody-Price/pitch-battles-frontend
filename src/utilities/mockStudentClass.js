export const mockStudentClass = {
  data: {
    id: "5",
    type: "class_dashboard",
    attributes: {
      name: "6th Grade Orchestra",
      class_key: "K5s5qCQfHTYYldIhDcmGQg",
      teacher: {
        data: {
          id: "6",
          type: "bare_user",
          attributes: {
            email: "relasine@gmail.com",
            first_name: "Kevin",
            last_name: "Simpson",
            avatar: 1,
            role: "teacher",
            level_one_fastest_time: 0,
            level_two_fastest_time: 0,
            level_three_fastest_time: 0,
            level_four_fastest_time: 0,
            total_games_played: 0
          }
        }
      },
      level_one_fastest_time: {
        score: 9401,
        user: {
          data: [
            {
              id: "1",
              type: "bare_user",
              attributes: {
                email: "simpsonkevinjohn@gmail.com",
                first_name: "Chet",
                last_name: "Manly",
                avatar: 11,
                role: "student",
                level_one_fastest_time: 9401,
                level_two_fastest_time: 15412,
                level_three_fastest_time: 21217,
                level_four_fastest_time: 27623,
                total_games_played: 24
              }
            }
          ]
        }
      },
      level_two_fastest_time: {
        score: 15412,
        user: {
          data: [
            {
              id: "1",
              type: "bare_user",
              attributes: {
                email: "simpsonkevinjohn@gmail.com",
                first_name: "Chet",
                last_name: "Manly",
                avatar: 11,
                role: "student",
                level_one_fastest_time: 9401,
                level_two_fastest_time: 15412,
                level_three_fastest_time: 21217,
                level_four_fastest_time: 27623,
                total_games_played: 24
              }
            }
          ]
        }
      },
      level_three_fastest_time: {
        score: 21217,
        user: {
          data: [
            {
              id: "1",
              type: "bare_user",
              attributes: {
                email: "simpsonkevinjohn@gmail.com",
                first_name: "Chet",
                last_name: "Manly",
                avatar: 11,
                role: "student",
                level_one_fastest_time: 9401,
                level_two_fastest_time: 15412,
                level_three_fastest_time: 21217,
                level_four_fastest_time: 27623,
                total_games_played: 24
              }
            }
          ]
        }
      },
      level_four_fastest_time: {
        score: 27623,
        user: {
          data: [
            {
              id: "1",
              type: "bare_user",
              attributes: {
                email: "simpsonkevinjohn@gmail.com",
                first_name: "Chet",
                last_name: "Manly",
                avatar: 11,
                role: "student",
                level_one_fastest_time: 9401,
                level_two_fastest_time: 15412,
                level_three_fastest_time: 21217,
                level_four_fastest_time: 27623,
                total_games_played: 24
              }
            }
          ]
        }
      },
      overall_fastest_time: {
        score: 33333,
        user: {
          data: [
            {
              id: "3",
              type: "bare_user",
              attributes: {
                email: "dmeskis@gmail.com",
                first_name: "Dylan",
                last_name: "Meskis",
                avatar: 11,
                role: "student",
                level_one_fastest_time: 13402,
                level_two_fastest_time: 24216,
                level_three_fastest_time: 1234134,
                level_four_fastest_time: 0,
                total_games_played: 8
              }
            }
          ]
        }
      },
      most_games: {
        games_played: 24,
        user: {
          data: [
            {
              id: "1",
              type: "bare_user",
              attributes: {
                email: "simpsonkevinjohn@gmail.com",
                first_name: "Chet",
                last_name: "Manly",
                avatar: 11,
                role: "student",
                level_one_fastest_time: 9401,
                level_two_fastest_time: 15412,
                level_three_fastest_time: 21217,
                level_four_fastest_time: 27623,
                total_games_played: 24
              }
            }
          ]
        }
      },
      most_badges: {
        badges: 12,
        user: {
          data: [
            {
              id: "1",
              type: "bare_user",
              attributes: {
                email: "simpsonkevinjohn@gmail.com",
                first_name: "Chet",
                last_name: "Manly",
                avatar: 11,
                role: "student",
                level_one_fastest_time: 9401,
                level_two_fastest_time: 15412,
                level_three_fastest_time: 21217,
                level_four_fastest_time: 27623,
                total_games_played: 24
              }
            }
          ]
        }
      }
    }
  }
};
