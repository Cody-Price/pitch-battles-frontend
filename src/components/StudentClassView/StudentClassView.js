import React, { Component } from "react";
import "./StudentClassView.css";
import LoadingAnimation from "../LoadingAnimation/LoadingAnimation";
import { studentClassFetch } from "../../utilities/fetchCalls";
import PropTypes from "prop-types";
import avatars from "../../utilities/avatars";

class StudentClassView extends Component {
  constructor() {
    super();

    this.state = {
      classStats: undefined,
      loading: true,
      error: false
    };
  }

  componentDidMount() {
    this.studentViewClassFetch();
  }

  studentViewClassFetch = async () => {
    try {
      const response = await studentClassFetch(this.props.webToken);
      console.log(response);
      this.setState({
        classStats: response.data.attributes,
        loading: false,
        error: false
      });
    } catch (error) {
      console.log(error);
      this.setState({ error: true });
    }
  };

  render() {
    return (
      <div className="student-class-view-wrapper">
        {!this.state.classStats && <LoadingAnimation animals="cat" />}
        {this.state.classStats && (
          <article className="student-class-view">
            <header className="student-class-view-header">
              <button
                className="class-view-back-btn"
                onClick={() => this.props.navigate("student dash")}
              >
                exit
              </button>
              <h1 className="student-class-view-class-name">
                {this.state.classStats.name}
              </h1>
              <h2 className="student-class-view-teacher-name">
                teacher -{" "}
                {this.state.classStats.teacher.data.attributes.first_name}{" "}
                {this.state.classStats.teacher.data.attributes.last_name}
              </h2>
            </header>

            <section className="class-view-fastest-times">
              {this.state.classStats.level_one_fastest_time.user.data[0] && (
                <h3 className="class-view-fastest-times-label">
                  fastest times
                </h3>
              )}
              <div className="class-view-fastest-times-row">
                {this.state.classStats.level_one_fastest_time.user.data[0] && (
                  <article className="class-view-fastest-time">
                    <h3 className="class-view-level">level one</h3>
                    <p className="class-view-fastest-time-number">
                      {this.state.classStats.level_one_fastest_time.score /
                        1000}
                    </p>
                    <p className="class-view-fastest-student">
                      {
                        this.state.classStats.level_one_fastest_time.user
                          .data[0].attributes.first_name
                      }{" "}
                      {
                        this.state.classStats.level_one_fastest_time.user
                          .data[0].attributes.last_name
                      }
                    </p>
                    <div
                      className={`avatar-class-view ${
                        avatars[
                          this.state.classStats.level_one_fastest_time.user
                            .data[0].attributes.avatar
                        ]
                      }`}
                    />
                  </article>
                )}
                {this.state.classStats.level_two_fastest_time.user.data[0] && (
                  <article className="class-view-fastest-time">
                    <h3 className="class-view-level">level two</h3>
                    <p className="class-view-fastest-time-number">
                      {this.state.classStats.level_two_fastest_time.score /
                        1000}
                    </p>
                    <p className="class-view-fastest-student">
                      {
                        this.state.classStats.level_two_fastest_time.user
                          .data[0].attributes.first_name
                      }{" "}
                      {
                        this.state.classStats.level_two_fastest_time.user
                          .data[0].attributes.last_name
                      }
                    </p>
                    <div
                      className={`avatar-class-view ${
                        avatars[
                          this.state.classStats.level_two_fastest_time.user
                            .data[0].attributes.avatar
                        ]
                      }`}
                    />
                  </article>
                )}
                {this.state.classStats.level_three_fastest_time.user
                  .data[0] && (
                  <article className="class-view-fastest-time">
                    <h3 className="class-view-level">level three</h3>
                    <p className="class-view-fastest-time-number">
                      {this.state.classStats.level_three_fastest_time.score /
                        1000}
                    </p>
                    <p className="class-view-fastest-student">
                      {
                        this.state.classStats.level_three_fastest_time.user
                          .data[0].attributes.first_name
                      }{" "}
                      {
                        this.state.classStats.level_three_fastest_time.user
                          .data[0].attributes.last_name
                      }
                    </p>
                    <div
                      className={`avatar-class-view ${
                        avatars[
                          this.state.classStats.level_three_fastest_time.user
                            .data[0].attributes.avatar
                        ]
                      }`}
                    />
                  </article>
                )}
                {this.state.classStats.level_four_fastest_time.user.data[0] && (
                  <article className="class-view-fastest-time">
                    <h3 className="class-view-level">level four</h3>
                    <p className="class-view-fastest-time-number">
                      {this.state.classStats.level_four_fastest_time.score /
                        1000}
                    </p>
                    <p className="class-view-fastest-student">
                      {
                        this.state.classStats.level_four_fastest_time.user
                          .data[0].attributes.first_name
                      }{" "}
                      {
                        this.state.classStats.level_four_fastest_time.user
                          .data[0].attributes.last_name
                      }
                    </p>
                    <div
                      className={`avatar-class-view ${
                        avatars[
                          this.state.classStats.level_four_fastest_time.user
                            .data[0].attributes.avatar
                        ]
                      }`}
                    />
                  </article>
                )}
                {this.state.classStats.overall_fastest_time.user.data[0] && (
                  <article className="class-view-fastest-time">
                    <h3 className="class-view-level">overall</h3>
                    <p className="class-view-fastest-time-number">
                      {this.state.classStats.overall_fastest_time.score / 1000}
                    </p>
                    <p className="class-view-fastest-student">
                      {
                        this.state.classStats.overall_fastest_time.user.data[0]
                          .attributes.first_name
                      }{" "}
                      {
                        this.state.classStats.overall_fastest_time.user.data[0]
                          .attributes.last_name
                      }
                    </p>
                    <div
                      className={`avatar-class-view ${
                        avatars[
                          this.state.classStats.overall_fastest_time.user
                            .data[0].attributes.avatar
                        ]
                      }`}
                    />
                  </article>
                )}
              </div>
            </section>
            <section className="most-games-badges-row">
              {this.state.classStats.most_games && (
                <article className="most-games game-badge">
                  <h3 className="most-games-badges-label">most games</h3>
                  {this.state.classStats.most_games.user.data[0].attributes
                    .first_name && (
                    <p className="class-view-most-student">
                      {
                        this.state.classStats.most_games.user.data[0].attributes
                          .first_name
                      }{" "}
                      {
                        this.state.classStats.most_games.user.data[0].attributes
                          .last_name
                      }{" "}
                      - {this.state.classStats.most_games.games_played}
                    </p>
                  )}
                  <div
                    className={`avatar-class-view ${
                      avatars[
                        this.state.classStats.most_games.user.data[0].attributes
                          .avatar
                      ]
                    }`}
                  />
                </article>
              )}
              {this.state.classStats.most_badges && (
                <article className="most-badges game-badge">
                  <h3 className="most-games-badges-label">most badges</h3>
                  {this.state.classStats.most_badges && (
                    <p className="class-view-most-student">
                      {
                        this.state.classStats.most_badges.user.data[0]
                          .attributes.first_name
                      }{" "}
                      {
                        this.state.classStats.most_badges.user.data[0]
                          .attributes.last_name
                      }{" "}
                      - {this.state.classStats.most_badges.badges}
                    </p>
                  )}
                  <div
                    className={`avatar-class-view ${
                      avatars[
                        this.state.classStats.most_badges.user.data[0]
                          .attributes.avatar
                      ]
                    }`}
                  />
                </article>
              )}
            </section>
          </article>
        )}
      </div>
    );
  }
}

export default StudentClassView;

StudentClassView.propTypes = {
  webToken: PropTypes.string,
  navigate: PropTypes.func
};
