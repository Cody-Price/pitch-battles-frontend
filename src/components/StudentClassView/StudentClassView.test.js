import React from "react";
import StudentClassView from "./StudentClassView";
import { mockStudentClass } from "../../utilities/mockStudentClass";
import Enzyme, { shallow } from "enzyme";
const EnzymeAdapter = require("enzyme-adapter-react-16");

Enzyme.configure({ adapter: new EnzymeAdapter() });

jest.mock("../../utilities/fetchCalls");

describe("StudentClassView", () => {
  let wrapper;
  let mockNavigate;

  beforeEach(() => {
    const mockWebToken = "awklefj";
    mockNavigate = jest.fn();
    wrapper = shallow(
      <StudentClassView navigate={mockNavigate} webToken={mockWebToken} />
    );
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe("studentViewClassFetch", () => {
    it("should call set state", async () => {
      await wrapper.instance().studentViewClassFetch();

      expect(wrapper.state().classStats).toEqual(
        mockStudentClass.data.attributes
      );

      expect(wrapper.state().loading).toEqual(false);
    });
  });

  describe("navitage", () => {
    it("should call navigate on click", () => {
      wrapper.find(".class-view-back-btn").simulate("click");

      expect(mockNavigate).toHaveBeenCalledWith("student dash");
    });
  });
});
