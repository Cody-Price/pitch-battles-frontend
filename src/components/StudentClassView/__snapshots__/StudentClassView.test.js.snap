// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudentClassView should match the snapshot 1`] = `
<div
  className="student-class-view-wrapper"
>
  <LoadingAnimation
    animals="cat"
  />
</div>
`;
